zerver/actions/message_edit.py

```diff
@@ -577,6 +567,9 @@ def user_info(um: UserMessage) -> dict[str, Any]:
         # mention_data is required if there's a content edit.
         assert mention_data is not None
 
+        # ResolveTopicRequest has no content edited.
+        assert not isinstance(message_edit_request, ResolveTopicRequest)
```

Comment:
Generally a positive assertion with the list of valid classes is what we want, rather than exclusions like this.